src/lifecycles/prop.helpers.js

```diff
@@ -2,10 +2,32 @@ import * as singleSpa from "../single-spa.js";
 import { mountParcel } from "../parcels/mount-parcel.js";
 import { assign } from "../utils/assign.js";
 import { isParcel, toName } from "../applications/app.helpers.js";
+import { formatErrorMessage } from "../applications/app-errors.js";
 
 export function getProps(appOrParcel) {
-  const result = assign({}, appOrParcel.customProps, {
-    name: toName(appOrParcel),
+  const name = toName(appOrParcel);
+  let customProps =
+    typeof appOrParcel.customProps === "function"
+      ? appOrParcel.customProps(name, window.location)
```

Comment:
I arrived at passing in the name and current location to the custom props function, as those might be helpful for those implementing "dynamic custom props." I am open to rethinking which arguments are good ones to pass to the customProps function. For single-spa layout, the location is especially important.

Reply:
Yeah that makes sense to me. I'm not 100% sure what the name will be used for but it seems silly to not provide it.